library/ajax/clinical_stats_by_demographics_report_ajax.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * library/ajax/claims_viewer_ajax.php Ajax file to process imported X12 filesgstat
+ * file adapted to present user activity log
+ * Copyright (C) 2012 Medical Information Integration <info@mi-squared.com>
+ * Copyright (c) 2018 Growlingflea Software <daniel@growlingflea.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ */
+ $fake_register_globals=false;
+ $sanitize_all_escapes=true;
+ $testing = true;
+
+
+require_once("../../interface/globals.php");
+require_once("$srcdir/sql.inc");
+require_once("$srcdir/formatting.inc.php");
+require_once("$srcdir/acl.inc");
+require_once("$srcdir/patient.inc");
+
+
+
+//Enter false if searching for users that aren't active
+function getUsersArray($active = true){
+
+    $users = array();
+    $query = "Select distinct(username) from users ";
+    $query .= $active = true ? "where active = 1" : " ";
+    $query .= " AND username != ''";
+    $result = sqlStatement($query);
+    while ($row = sqlFetchArray($result)) {
+        array_push($users, $row['username']);
+
+    }
+    return $users;
+}
+
+function ifTestingTrue($testing){
+
+    if($testing)
+        return " Limit 0,100";
+    else return "";
+
+}
+
+
+if($_POST['func']=="get_all_diags_data")
+{
+    $response['data'] = array();
+
+    $query = "SELECT patient_data.pid, issue_encounter.encounter, form_encounter.date, type, lists.title, lists.diagnosis, activity, dob, city, state, status, sex, race, ethnicity ";
+    $query .= "FROM `issue_encounter` ";
+    $query .= "join lists on list_id = lists.id ";
+    $query .= "join patient_data on lists.pid = patient_data.pid ";
+    $query .= "join form_encounter on issue_encounter.encounter = form_encounter.encounter";
```

Comment:
@growlingflea If you want to do paging to drastically improve performance, you can add LIMIT start, start+length. Those variables are passed to the server by datatables. check out docs: the https://datatables.net/manual/server-side